components/async-operation-status-mgt/org.wso2.carbon.identity.framework.async.operation.status.mgt/src/main/java/org/wso2/carbon/identity/framework/async/operation/status/mgt/internal/dao/impl/AsyncOperationOperationStatusMgtDAOImpl.java

```diff
@@ -0,0 +1,584 @@
+                    return null;
+                }
+                UnitOperationDO record = new UnitOperationDO();
+                record.setUnitOperationId(resultSet.getString(1));
+                record.setOperationId(resultSet.getString(2));
+                record.setOperationInitiatedResourceId(resultSet.getString(3));
+                record.setTargetOrgId(resultSet.getString(4));
+                record.setUnitOperationStatus(resultSet.getString(5));
+                record.setStatusMessage(resultSet.getString(6));
+                record.setCreatedTime(Timestamp.valueOf(resultSet.getString(7)));
+                return record;
+            }, namedPreparedStatement -> {
+                namedPreparedStatement.setString(UNIT_OPERATION_ID, unitOperationId);
+                namedPreparedStatement.setString(INITIATED_ORG_ID, requestInitiatedOrgId);
+            });
+            return unitOperationRecord;
+        } catch (DataAccessException e) {
+            throw handleServerException(ERROR_WHILE_RETRIEVING_ASYNC_OPERATION_STATUS_UNIT, e);
+        }
+    }
+
+    private void deleteOldOperationalData(Connection connection, String correlationId, String operationType,
+                                          String operationSubjectId) throws SQLException {
+
+        try (NamedPreparedStatement statement = new NamedPreparedStatement(connection,
+                DELETE_RECENT_OPERATION_RECORD)) {
+            statement.setString(OPERATION_TYPE, operationType);
+            statement.setString(SUBJECT_ID, operationSubjectId);
+            statement.setString(CORRELATION_ID, correlationId);
+            statement.executeUpdate();
+        }
+    }
+
+    private UnitOperationStatusCount getUnitOperationStatusCount(String operationId, String requestInitiatedOrgId)
+            throws  DataAccessException {
+
+        UnitOperationStatusCount countObj = new UnitOperationStatusCount();
+        NamedJdbcTemplate namedJdbcTemplate = Utils.getNewTemplate();
+
+        namedJdbcTemplate.executeQuery(GET_UNIT_OPERATION_STATUS_COUNT, (resultSet, rowNumber) -> {
+
+            String status = resultSet.getString(1);
+            int count = resultSet.getInt(2);
+
+            if (StringUtils.equals(status, SUCCESS.toString())) {
+                countObj.setSuccess(count);
+            } else if (StringUtils.equals(status, FAILED.toString())) {
+                countObj.setFailed(count);
+            } else if (StringUtils.equals(status, PARTIALLY_COMPLETED.toString())) {
+                countObj.setPartiallyCompleted(count);
+            }
+            return null;
+        },
+        namedPreparedStatement -> {
+            namedPreparedStatement.setString(OPERATION_ID, operationId);
+            namedPreparedStatement.setString(INITIATED_ORG_ID, requestInitiatedOrgId);
+        });
+        return countObj;
+    }
+
+    private FilterQueryBuilder buildFilterQuery(List<ExpressionNode> expressionNodes, String attributeUsedForCursor)
```

Comment:
Let's move all building filter query related private methods to a separate util class. e.g., `FilterQueryBuilderUtil`